patroni/postgresql.py

```diff
@@ -173,14 +180,32 @@ def pg_ctl(self, cmd, *args, **kwargs):
         pg_ctl = [self._pgcommand('pg_ctl'), cmd]
         if cmd in ('start', 'stop', 'restart'):
             pg_ctl += ['-w']
-            timeout = self.config.get('pg_ctl_timeout')
+            timeout = kwargs.pop('timeout', None)
+            timeout = self.config.get('pg_ctl_timeout', timeout)
             if timeout:
                 try:
                     pg_ctl += ['-t', str(int(timeout))]
                 except Exception:
                     logger.error('Bad value of pg_ctl_timeout: %s', timeout)
         return subprocess.call(pg_ctl + ['-D', self._data_dir] + list(args), **kwargs) == 0
 
+    def pg_isready(self):
+        """Runs pg_isready to see if PostgreSQL is accepting connections.
+
+        :returns: 'ok' if PostgreSQL is up, 'reject' if starting up, 'no_resopnse' if not up."""
+
+        cmd = [self._pgcommand('pg_isready'),
+               '-h', self._local_address['host'],
+               '-p', self._local_address['port'],
+               '-d', self._database,
+               '-U', self._superuser]
```

Comment:
self._superuser is a dict object:
```python
{'username': 'postgres', 'password': 'blabla'}
```

So far it was not mandatory that `superuser` object should have username and password defined.
In case if there is no username, libpq is taking `PGUSER` from env or just `USER`. That means `-U` argument can be optional... 

Also pg_hba.conf may require password even for local connections. That means we should write password (if it is available) into `pgpass` file and export pgpass filename as `PGPASSFILE` environment variable. We already have `write_pgpass` method doing all this things.